### ISSUE TITLE
paru -Sc always cleans devel packages

### ISSUE THREAD

**Issue by conlogic** [Morganamilo/paru#314]
### Affected Version
`paru -V`: paru v1.5.0.r4.g9d9cdb1 - libalpm v12.0.2

### Description
When I issue an `paru -Sc` and answer `Y` to

```sh
Clone Directory: /var/cache/paru/build
:: Do you want to remove all other AUR packages from cache? [Y/n]:
```
all built `*-git` packages are deleted from paru's cache - even all installed ones. By the way, this is not affected by the value of `CleanMethod` in the used `pacman.conf` file.

**Comment by Morganamilo** [comment]
That's what -Sc. What do you expect it to do?

**Comment by conlogic** [comment]
I expect for devel packages a similar behavior like for stable packages - both for repository and AUR packages: that the last version of the package is kept in paru's cache. To be a bit more specific: At least if in the used `pacman.conf` is set as:

```ini
CleanMethod = KeepInstalled
```
the installed version of a `-git` packages is kept in paru's cache.

**Comment by Morganamilo** [comment]
Right, you didn't mention the clean method in the original issue. There's not code to treat devel packages differently. Are you sure the issue is devel packages always get cleaned and not some other factor?

**Comment by conlogic** [comment]
> Are you sure the issue is devel packages always get cleaned and not some other factor?

Of course I am not sure, but when using the scenario described above in the original issue, i.e.:
* Calling `paru -Sc`;
* Answering `Y` to the third question mentioned above,
I always get the built packages for `-git` packages cleaned but all stable packages - even from AUR - handled according to the `CleanMethod` value in the used `pacman.conf`.

Do you get different results for `-git` packages in this scenario?

**Comment by M1T0zd** [comment]
I've just done exactly this and it has deleted the `sudo` package (and only sudo seemingly for some reason). Now without sudo and without knowing my root password, I am locked out of root privileges.

Currently trying to solve this without reinstalling Arch, but I'm not sure if it's possible. Quite a headache.

I do hope accidentally deleting `base-devel` packages would be made harder to prevent others from facing a similar fate. 1+ for this issue.

**Comment by Morganamilo** [comment]
Not sure what you're on about as this issue is about clearing the paru cache.

**Comment by M1T0zd** [comment]
Honestly, I'm not knowledgeable enough to tell you exactly why or what. But I'm quite sure this happened after doing `paru -Sc`.

I just saw this line in the installation section of the readme however: `sudo pacman -S --needed base-devel`
I think that would have prevented this problem? But since I installed paru via yay, I haven't run this command.

But anyways. If there is something I can do to make more clear to you, let me know.
